### PR TITLE
ec2: add aws_spot_fleet_request

### LINKED ISSUE #185: Moving spot (1) resources to v1beta1 version
Moving 1 resources in the `spot` group:
- [ ] [`aws_spot_fleet_request`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/spot_fleet_request)

### PR DESCRIPTION
**Pull request by anastasiia-kvas**
### Description of your changes
Adding new resources from group ec2: 
- [x] [`aws_spot_fleet_request`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/spot_fleet_request)

Fixes #185 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
`aws_spot_fleet_request` - manually and **uptest** https://github.com/upbound/provider-aws/actions/runs/3683986036
Screenshot 2022-12-12 at 11 46 38_true_true

### CODE REVIEW

**Comment by dverveiko** [comment]
/test-examples="examples/ec2/spotfleetrequest.yaml"

**Comment by sergenyalcin** [comment]
@anastasiia-kvas thanks for this PR. Could you please rebase the branch? `CI / report-breaking-changes` job fails with this error `fetch_depth to a number higher than 40.`

**Comment by sergenyalcin** [comment]
/test-examples="examples/ec2/spotfleetrequest.yaml"

File: examples/ec2/spotfleetrequest.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+metadata:
+  annotations:
+    meta.upbound.io/example-id: ec2/v1beta1/spotfleetrequest
+  labels:
+    testing.upbound.io/example-name: cheap_compute
+  name: cheap-compute
+spec:
+  forProvider:
+    allocationStrategy: capacityOptimized
+    iamFleetRole: arn:aws:iam::${data.aws_account}:role/aws-service-role/spotfleet.amazonaws.com/AWSServiceRoleForEC2SpotFleet
```
Comment (sergenyalcin) [inline examples/ec2/spotfleetrequest.yaml:12]: For Account ID we need to use the following key for reading the value from the data source: `${data.aws_account_id}`

```suggestion
    iamFleetRole: arn:aws:iam::${data.aws_account_id}:role/aws-service-role/spotfleet.amazonaws.com/AWSServiceRoleForEC2SpotFleet
```

**Comment by steperchuk** [comment]
/test-examples="examples/ec2/spotfleetrequest.yaml"

**Review by sergenyalcin — APPROVED** [review]
Thanks @anastasiia-kvas LGTM!

**Pull request merged**